crates/air-lib/air-parser/src/ast/values/impls.rs

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2021 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use super::*;
+use air_lambda_parser::LambdaAST;
+use air_lambda_parser::ValueAccessor;
+
+impl<'i> ScalarWithLambda<'i> {
+    pub fn new(name: &'i str, lambda: Option<LambdaAST<'i>>) -> Self {
+        Self { name, lambda }
+    }
+
+    pub(crate) fn from_raw_lambda(name: &'i str, lambda: Vec<ValueAccessor<'i>>) -> Self {
+        let lambda = unsafe { LambdaAST::new_unchecked(lambda) };
+        Self {
+            name,
+            lambda: Some(lambda),
+        }
+    }
+}
+
+impl<'i> StreamWithLambda<'i> {
+    pub fn new(name: &'i str, lambda: Option<LambdaAST<'i>>) -> Self {
+        Self { name, lambda }
+    }
+
+    #[allow(dead_code)]
+    pub(crate) fn from_raw_lambda(name: &'i str, lambda: Vec<ValueAccessor<'i>>) -> Self {
+        let lambda = unsafe { LambdaAST::new_unchecked(lambda) };
```

Comment:
It's not safe and is intended to use only in tests (that why `pub(crate)`), but I'll add a comment about that since all other such code have a comment.